implement `net.cidr_is_valid` builtin

tjons opened the pull request:
@anakrish I've been working on implementing the `net.cidr_is_valid` builtin. It currently is complete for IPv4, and so I wanted to try to get some early feedback while I finish support for IPv6.

I'm still quite new to Rust, and so please feel free to give any pointers or suggestions you think I might be missing :crab:

tjons commented:
@microsoft-github-policy-service agree

anakrish reviewed (commented):
Hey @tjons,
Thanks for the contribution. Overall looks good to me. I've provided some suggestions.

tjons commented:
@anakrish thanks for the feedback! I'll update this and finish ipv6 this week. Do let me know your thoughts on https://github.com/microsoft/regorus/pull/422#discussion_r2197652414, which will either dramatically simplify the implementation or take a bit longer :)

tjons commented:
@anakrish sorry for the wait, addressed all the feedback! Let me know if anything else is needed for this builtin, I'll get started on the rest this week.

anakrish commented:
Hey @tjons, Overall looks good to me.

There are a bunch of format string related failures. https://github.com/microsoft/regorus/actions/runs/16734832282/job/47424635159?pr=422

Can you also add `v0/netcidrisvalid` and `v1/netcidrisvalid` to https://github.com/microsoft/regorus/blob/main/tests/opa.passing to ensure that tests from OPA testsuite are run?

anakrish commented:
@tjons No worries. I truly glad that you are contributing.
